src/literals.hpp

```diff
@@ -0,0 +1,14 @@
+#include <stdint.h>
+
+/*
+
+Cross platform custom literals.  Clang/GCC demand the underscore for 
+integer literals.
+
+*/
+
+constexpr int16_t operator "" _i16(unsigned long long int lit){return ((int16_t) lit);}
```

Comment:
Unsure if the unsigned long long int parameter breaks the signage.  The standard [permits](https://stackoverflow.com/a/16596909/4476082) only a select few types, maybe ```long double``` would be better?  Not sure if it even matters.